### PR TITLE
Update beam_fn_api.proto to specify allowed split points.

### PR DESCRIPTION
**Pull request by ananvay**
allowed_split_points will allow the runner to specify indices where the SDK may split.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by ananvay** [comment]
R:@robertwb,@lukecwik

File: model/fn-execution/src/main/proto/beam_fn_api.proto [hunk 1]
```diff
@@ -765,6 +765,9 @@ message ProcessBundleSplitRequest {
 
     // TODO(SDF): Allow providing weights rather than sizes.
     // TODO(SDF): Allow specifying allowed/preferred split points.
```
Comment (robertwb) [inline model/fn-execution/src/main/proto/beam_fn_api.proto]: Remove this TODO.

File: model/fn-execution/src/main/proto/beam_fn_api.proto [hunk 2]
```diff
@@ -765,6 +765,9 @@ message ProcessBundleSplitRequest {
...
+
+    // A set of allowed indices where the SDK may split.
+    repeated int64 allowed_split_points = 3;
```
Comment (robertwb) [inline model/fn-execution/src/main/proto/beam_fn_api.proto]: Put right after fraction_of_remainder. Also note that if this is empty, there is no constraint.

Comment (ananvay) [reply model/fn-execution/src/main/proto/beam_fn_api.proto]: done

**Comment by ananvay** [comment]
Thanks Robert! PTAL.

**Comment by lukecwik** [comment]
Did we mean to merge this into a release branch?

**Comment by ananvay** [comment]
Yeah, we need this in the master branch, not the release.

**Comment by angoenka** [comment]
Correct pull request to master https://github.com/apache/beam/pull/8476

**Comment by robertwb** [comment]
Oh, I just naturally assumed this was a PR against master. Sorry.

**Pull request merged**